samples/nrf_desktop/src/modules/hid_state.c

```diff
@@ -980,6 +980,7 @@ static bool event_handler(const struct event_header *eh)
 				connect_subscriber(event->id, true);
 				break;
 			case USB_STATE_DISCONNECTED:
+			case USB_STATE_SUSPENDED:
```

Comment:
This will fix the bug we see now but it is not the right thing to do. The main problem we see is that from the point of view of hid state USB can be plugged in multiple times while in fact nothing change on the USB socket (i.e. device remains plugged). This happens as we broadcast state changes from usb_state module and powered will go from both cable connection and resume or reset. Note that thing you do here is fixing case when resume is sent but not fixing the reset.

In my opinion there are two ways for fixing the bug properly:
- Broadcast USB_STATE_POWERED only when socket is plugged in. If needed we could create an additional state that would mean powered but with no plug in/out. The pros is usb state is managed only in dedicated module (other module trust state is correct - we should assert there is no usb subscriber in fact). The dis is that the additional state in usb module is not really needed. I actually wanted to implement it this way.
- Or we can simply check in hid_state if USB was already plugged (assuming that we don't really care about the usb state but we care only if usb was plugged). You could call `get_subscriber` and add subscriber only if it was not present. This makes hid state aware of the fact that powered means plugged in or reset/resumed which is dis. But pros is that change would be very simple.

Reply:
Fixed. Added an additional state in usb_state to be able to ignore additional USB events.